text/0000-kpack-donation-to-cnb.md

```diff
@@ -0,0 +1,355 @@
+
+# Risks
+
+- It's not clear how to handle the budget required to finance the infrastructure to rebuild the CI/CD pipelines on CNCF CNB infrastructure.
+- Evaluate any legal requirement from [CNCF](https://www.cncf.io) that must be fulfilled before accepting the project into the [CNB](https://buildpacks.io/) ecosystem.
+- Relying on the approval of [Component Maintainer Role RFC](https://github.com/buildpacks/rfcs/pull/234) to guarantee current [kpack](https://github.com/pivotal/kpack/) maintainers could keep supporting the project after the donation.
+
+# Drawbacks
+
+Why should we _not_ do this?
+
+- If the [CNB](https://buildpacks.io/) team expects to implement a different kind of integration with [Kubernetes](https://kubernetes.io/), then accepting the donation of [kpack](https://github.com/pivotal/kpack/) could conflict with that strategy.
+- Another component to maintain which requires additional context and expertise in [Kubernetes](https://kubernetes.io/).
+
+# Alternatives
+
+- What other designs have been considered?
+  - [VMware](https://www.vmware.com/) could continue to control the project, but it doesn't help on increase the adoption because it remains as a single-vendor driven project
+  - [VMware](https://www.vmware.com/) could donate [kpack](https://github.com/pivotal/kpack/) to the [Continuous Delivery Foundation](https://cd.foundation/), but [CNB](https://buildpacks.io/) presents a natural home for [kpack](https://github.com/pivotal/kpack/)  (it is an implementation of the platform specification)
+  - [VMware](https://www.vmware.com/) could create a new [CNCF](https://www.cncf.io/) project and move all [kpack](https://github.com/pivotal/kpack/) resources to it, but in this case it would need to undergo as a sandbox project for example.
+
+- Why is this proposal the best?
+
+[kpack](https://github.com/pivotal/kpack/) is a mature Kubernetes-native tool that leverages buildpacks and is used in production environments. The project's maintainers and contributors possess valuable technical and user context, derived from developing [kpack](https://github.com/pivotal/kpack/) and integrating feedback from users utilizing [CNB](https://buildpacks.io/) concepts when presented as part of Kubernetes resources.
+
+- What is the impact of not doing this?
+
+The [CNB](https://buildpacks.io/) community would have to develop from scratch any kind of integration with the Cloud Native Ecosystem to satisfy the project goals.
+
+**Prior Art**
+
+- Guidelines for accepting component-level contributions [RFC #143](https://github.com/buildpacks/rfcs/pull/143)
+- Component Maintainer Role [RFC #234](https://github.com/buildpacks/rfcs/pull/234)
+- Proposal to move CNCF slack [RFC #198](https://github.com/buildpacks/rfcs/pull/198)
+
+# Unresolved Questions
```

Comment:
I like to see the following in this RFC:

* Who are the vendors/companies contributing to kpack today? Who is on the core team (is it VMware only, or diverse)? 
* Who are team members and maintainers (like a specific list of people) that would be joining CNB as component maintainers? Are there any other roles they will join as?

Reply:
I can include the list of maintainers from VMware side, which are willing to be more involved with CNB. Also, I can include some stats about the PRs contributions from VMware / non VMWare companies.

Other vendors/companies contributing: **Bloomberg** and **WP-engine** 
- @samj1912 / @AidanDelaney in a scenario of a donation of Kpack, are you planning to become maintainers? or does Bloomberg planning to invest more in the project?

Reply:
I personally would be interested yes. We will at the very least continue to sustain our existing investment into the kpack project. The investment might also grow.

Reply:
I am working on updating the contributor's stats, the last ones I have are 4 months old, but they reflect that most of the contributions come from VMware, but actually this is one of the reasons for proposing the donation. If I am not wrong, companies that are probably using kpack are not willing to invest or contribute more to the project because it is not vendor-neutral, it is a VMware open-source project, like Bloomberg. Even though this seems like a risk (from CNB's perspective) it is actually a step forward to encourage the community to get more involved. Also, from VMware's side, the investment does not seem to be reduced because kpack is an important piece of Tanzu's portfolio